kfp_tests/run_integration_tests.py

```diff
@@ -0,0 +1,84 @@
+import argparse
+from os import listdir
+from os.path import isfile, join
+from subprocess import run, PIPE
+
+import kfp
+
+from metaflow.util import get_username
+from metaflow.metaflow_config import KFP_SDK_API_NAMESPACE
+
+
+"""
+From the root of this project, run: 
+`python3 -m kfp_tests.run_integration_tests --dir_path [dir_path]
+    --namespace [namespace] --userid [userid]`
+
+This script runs all the flows in the `sample_flows` directory. It creates
+each kfp run, waits for the run to fully complete on KFP, and prints whether
+or not the run was successful. 
+
+The 3 argparse arguments are optional and should not be needed, but are provided 
+if you want to test for another namespace or from another user's account.
+
+Arguments:
+- flow_dir_path -- the directory where the sample flows are located. 
+    This should only change if you're not running this script from 
+    the root directory.
+- namespace -- the namespace on KF where the runs are created.
+    The default is obtained from your metaflow config file.
+- userid -- your userid for KF. The default is also 
+    obtained from your metaflow config file.
+"""
+
+
+def parse_run_id(output):
+    run_id_flag = False
+    run_id = ""
+    for i, char in enumerate(output):
+        if i >= 7 and output[i - 7 : i] == "run_id:":
+            run_id_flag = True
+        if run_id_flag and char == "\n":
+            return run_id
+        if run_id_flag:
+            run_id += char
+    return -1
+
+
+def obtain_flow_file_paths(flow_dir_path):
+    file_paths = [f for f in listdir(flow_dir_path) if isfile(join(flow_dir_path, f))]
+    file_paths.remove("__init__.py")  # __init__.py needed to establish Python module
+    return file_paths
+
+
+def test_sample_flows(args):
+    flow_file_paths = obtain_flow_file_paths(args.flow_dir_path)
+    kfp_client = kfp.Client(namespace=args.namespace, userid=args.userid)
+    for flow_file_path in flow_file_paths:
```

Comment:
Will some exception be thrown for some test cases? Maybe we can catch them and keep running other test files.

Reply:
@cloudw are you referring to test cases which are expected to fail? If they are to fail in KFP, that should be caught after the `wait_for_run_completion` step, but I didn't deal with that case yet. Is it necessary?

Reply:
@cloudw, essentially there are two types of issues here:
1) Compile time issues. These occur if the pipeline run couldn't be created. These are immediately caught with the if statement: `if run_id != -1`, as a correctly created run will have a valid `run_id`.
2) Run time issues. The occur while the run is happening, and if they happen, the value of `run_response["run"]["status"]` will not be `Succeeded`.
Are there any other cases you're thinking of?

Reply:
Test runner in general should be resilient to exceptions. It's hard to expect what code runs and how they fail in the future.

In subprocess you have `check=True` which could [raise exception on non-zero exit code from subprocess](https://docs.python.org/3/library/subprocess.html). If your python execution fails with exception (maybe someone break metaflow code, or we decide to raise errors instead of returning -1), the return code will be non-zero.

Reply:
@cloudw I will remove the `check=True` so the execution doesn't result in an exception within the testing code. Now, in PyTest, when the process exits abnormally, a nonzero return code is returned (failing a test), but no exception is thrown which might affect the testing procedure.